### PR TITLE
DOCSP-35308 Add Atlas Stream Connection Task

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## DESCRIPTION

Adding a stream processing task to the VSCode connection page. Also adding the `objects.inv` for Atlas docs to intersphinx for detail refs to stream processing topics.

## STAGING

https://preview-mongodbianfmongodb.gatsbyjs.io/mongodb-vscode/DOCSP-35308/connect/#connect-to-atlas-stream-processing

## JIRA

https://jira.mongodb.org/browse/DOCSP-35308

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=65b17686f394066377650d8f

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

**Review by Anemy — APPROVED** [review]

**Comment by Anemy** [comment]
@shaketbaby

File: source/connect.txt [hunk 1]
```diff
@@ -141,6 +141,59 @@ the shell connected to your selected deployment.
 
    :mongosh:`Perform CRUD Operations in the MongoDB Shell </crud>`
 
+
+.. _vsce-connect-task-atlas-streams:
+
+Create a Connection to Atlas Stream Processing
```
Comment (jmd-mongo) [inline source/connect.txt:147]: [suggestion]

Consider making this even more direct: "Connect to Atlas Stream Processing".

File: source/connect.txt [hunk 2]
```diff
@@ -141,6 +141,59 @@ the shell connected to your selected deployment.
...
+      An example SPI connection string:
+
+      .. code-block::
+
+         mongodb://user1:password1@atlas-stream-xxxxxxxxxxxxxxxxxxxxxxxx-yyyyy.virginia-usa.a.query.mongodb.net/?authSource=admin&readPreference=primary&ssl=true&directConnection=true
+
+   .. step:: Open a new |vsce| playground
+
+      After successfully connecting to your Atlas Stream, on the 
+      :guilabel:`Playgrounds` pane click :guilabel:`Create New Playground`.
```
Comment (jmd-mongo) [inline source/connect.txt:188]: [suggestion]

Consider re-ordering this like: "After successfully connecting to your Atlas Stream, click :guilabel:`Create New Playground`  on the  :guilabel:`Playgrounds` pane."

**Review by jmd-mongo — COMMENTED** [review]
Thanks for these updates, @ianf-mongodb! I've left a few suggestions for your consideration.

Thanks!
Joe

File: source/connect.txt (hunk 1 above)
Comment (ianf-mongodb) [reply source/connect.txt:147]: Done.

File: source/connect.txt (hunk 2 above)
Comment (ianf-mongodb) [reply source/connect.txt:188]: Updated.

**Review by jmd-mongo — APPROVED** [review]
Thanks for these updates, @ianf-mongodb! This one LGTM.

Thanks,
Joe

File: source/connect.txt (hunk 1 above)
Comment (RWaltersMA) [reply source/connect.txt]: Can we add how to get the connection string?  

You can use MongoDB for VS Code to connect to 
[Atlas Stream Processing](https://www.mongodb.com/products/platform/atlas-stream-processing)
 by providing a Stream Processing Interface (SPI) connection string.

To obtain the connection string navigate to the desired Stream Processor Instance, click Connect and select "MonfoDB for VSCode"

image

Comment (ianf-mongodb) [reply source/connect.txt]: Hi @RWaltersMA do you know ASP will be public? Currently, when I go to cloud.mongodb.com Stream Processing I can only request private preview access to the feature I added a link to the Stream processing docs as well. Should we add a note here that ASP is currently in private preview?

**Pull request merged**